scalafix-reflect/src/main/scala-3/scalafix/internal/reflect/RuleCompiler.scala

```diff
@@ -41,7 +40,7 @@ class RuleCompiler(
 
   private val compiler: Compiler = new Compiler()
   private val classLoader: AbstractFileClassLoader =
-    new AbstractFileClassLoader(target, this.getClass.getClassLoader)
```

Comment:
When the target is virtual (which is the case if no argument is provided like in `ScalafixToolbox`)
https://github.com/scalacenter/scalafix/blob/ac41300fe922ac27931064fd881272b927c3c72e/scalafix-reflect/src/main/scala-3/scalafix/internal/reflect/RuleCompiler.scala#L32
is creating a `dottyTargetDirectory` (used as compiler output) completely disconnected from `target` (used to try to classload later): they only share a name, but any memory space.

This commits fixes the problem before further refactoring to get rid of the line above.